qiskit_aer/quantum_info/states/aer_statevector.py

```diff
@@ -0,0 +1,273 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2017, 2019, 2020, 2021, 2022.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""
+Statevector quantum state class.
+"""
+import copy
+import numpy as np
+
+from qiskit.circuit import QuantumCircuit, Instruction
+from qiskit.quantum_info.states import Statevector
+from qiskit.quantum_info.operators.operator import Operator
+
+from qiskit_aer import AerSimulator
+from .aer_state import AerState
+from ...backends.aerbackend import AerError
+
+
+class AerStatevector(Statevector):
+    """AerStatevector class"""
+
+    def __init__(self, data, **configs):
+        """Initialize a statevector object.
+
+        Args:
+            data (np.array or list or Statevector or QuantumCircuit or qiskit.circuit.Instruction):
+                Data from which the statevector can be constructed. This can be either a complex
+                vector, another statevector or a ``QuantumCircuit`` or ``Instruction``
+                (``Operator`` is not supportted in the current implementation).  If the data is
+                a circuit or instruction, the statevector is constructed by assuming that all
+                qubits are initialized to the zero state.
+            configs (kwargs): configurations of ``AerState``. All the keys are handeld as string
```

Comment:
This either needs explaining here, or `AerState` needs to be added to the documentation build, and its documentation will need expanding (for example, to example the configuration).  Otherwise there's no way to find out what the available configuration options are other than reading the code.

What if somebody passes a `configs` dictionary that contains `"method": "density_matrix"`?

Reply:
good point. configurations are the same with `AerSimulator`. `statevector` and `matrix_product_state` are only available now.